lib/Myriad/Commands.pm

```diff
@@ -63,36 +65,29 @@ async method service (@args) {
 
 async method rpc ($rpc, @args) {
     try {
-        my $response = await $myriad->rpc_client->call_rpc($myriad->config->service_name->as_string, $rpc, @args);
+        my $service = Myriad::Service::Remote->new(myriad => $myriad, service_name => $myriad->registry->make_service_name($myriad->config->service_name->as_string));
+        my $response = await $service->call_rpc($rpc, @args);
         $log->infof('RPC response is %s', $response);
     } catch ($e) {
         $log->warnf('RPC command failed due: %s', $e);
     }
 }
 
 async method subscription ($stream, @args) {
-    my $service_name = $myriad->config->service_name->as_string;
-    $log->infof('Subscribing to: %s | %s | %s', $service_name, $stream);
-    my $sink = $myriad->ryu->sink(
-        label => "receiver:$stream",
-    );
-    $myriad->subscription->create_from_sink(
-        sink    => $sink,
-        channel => $stream,
-        client  => ref($self) . '/' . 'SUB_COMMAND',
-        from    => $service_name,
-    );
+    my $service = Myriad::Service::Remote->new(myriad => $myriad, service_name => $myriad->registry->make_service_name($myriad->config->service_name->as_string));
+    $log->infof('Subscribing to: %s | %s | %s', $service->service_name, $stream);
 
-    $sink->source->each(sub {
+    $service->subscribe($stream, 'SUB_COMMAND')->each(sub {
```

Comment:
`SUB_COMMAND`...? what's this and how do I convince you to get rid of it? :smile_cat:

Reply:
so that is the `consumer name` in for the consumer group in the `Myriad::Service::Implemeation` we give it the service name, not sure what we should name it in case of the top level command.